src/NUnitFramework/framework/Internal/Commands/OneTimeSetUpCommand.cs

```diff
@@ -75,15 +77,18 @@ public override TestResult Execute(TestExecutionContext context)
                     Test.Fixture = _suite.Fixture;
                 }
 
-                for (int i = _setUpTearDown.Count; i > 0; )
-                    _setUpTearDown[--i].RunSetUp(context);
+                if (!(suiteResult.Test is TestFixture && !suiteResult.Test.HasChildren))
```

Comment:
Excellent. I assume #1962 covers the concerns I had in https://github.com/nunit/nunit/pull/1940#issuecomment-270411026?